proxy/src/control/destination/endpoint.rs

```diff
@@ -1,34 +1,32 @@
-use futures_watch;
-use std::{cmp, hash, net::SocketAddr};
+use std::net::SocketAddr;
 
 use telemetry::metrics::DstLabels;
 
-pub type DstLabelsWatch = futures_watch::Watch<Option<DstLabels>>;
 
 /// An individual traffic target.
 ///
 /// Equality, Ordering, and hashability is determined soley by the Endpoint's address.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
 pub struct Endpoint {
     address: SocketAddr,
-    dst_labels: Option<DstLabelsWatch>,
+    dst_labels: Option<DstLabels>,
 }
 
 // ==== impl Endpoint =====
 
 impl Endpoint {
-    pub fn new(address: SocketAddr, dst_labels: DstLabelsWatch) -> Self {
+    pub fn new(address: SocketAddr, dst_labels: Option<DstLabels>) -> Self {
```

Comment:
We should consider making a future PR (not changing this one) that replaces `Option<DstLabels>` with `DstLabels` that can hold an empty set of labels. I'm not sure we even need to have any kind of optimization for the case of empty labels because that's supposed to be relatively rare, IIUC.

Reply:
It's probably a good idea to move the  `Option` to inside of `DstLabels`; I think covering the case of empty labels _is_ important, since endpoints found via DNS now also have `Metadata` associated with them (so they can have a `supports_tls: false` flag in #1008).